### PR TITLE
Deprecate GeoPolygon query in favour of GeoShape query.

### PR DESCRIPTION
**Pull request by iverase**
Following the development in #48928, GeoShape queries in more powerful that GeoPolygon queries and it makes little sense having both. This commit deprecates GeoPolygon queries.

### CODE REVIEW

File: docs/reference/query-dsl/geo-polygon-query.asciidoc [hunk 1]
```diff
@@ -80,6 +84,7 @@ GET /_search
   }
 }
 --------------------------------------------------
+// TEST[warning:Deprecated field [geo_polygon] used, replaced by [[geo_shape] query where polygons are defined in geojson or wkt]]
```
Comment (mayya-sharipova) [inline docs/reference/query-dsl/geo-polygon-query.asciidoc:87]: Is the `geo_polygon` is a different query from `geo_shape`? 
Or is it the same query and it was just renamed from `geo_polygon` to `geo_shape`?

This warning looks like we just renamed the query.  I thought we use this kind of warning when we rename the fields.

Comment (iverase) [reply docs/reference/query-dsl/geo-polygon-query.asciidoc:87]: It is a different query that has the same functionality (and much more) that the geo_polygon query.  I try to follow the same strategy as here: https://github.com/elastic/elasticsearch/pull/42619.

Comment (mayya-sharipova) [reply docs/reference/query-dsl/geo-polygon-query.asciidoc:87]: @iverase Thanks for the clarification, I was not aware that we use `withAllDeprecated` also with queries!

**Review by mayya-sharipova — APPROVED** [review]
@iverase Thanks, this LGTM.
One clarification, is `geo_polygon` query will be removed in v 8.x?  Or we will still keep it in 8.x for Rest API version compatibility in case 7.x clients want to access it?

**Comment by iverase** [comment]
The plan is to keep it in 8.x and remove it in 9.0.

**Pull request merged**